awx/main/tests/functional/test_dispatch.py

```diff
@@ -412,6 +419,16 @@ def test_workflow_does_not_reap(self):
         i.save()
         j = WorkflowJob(status='running', execution_node='awx')
         j.save()
-        reaper.reap(i)
+        reaper.reap(i, ref_time=now)
 
         assert WorkflowJob.objects.first().status == 'running'
+
+    def test_should_not_reap_new(self):
```

Comment:
Considering how nuanced and deliberate the ordering of commands in this test are, and how it's tied back to a very highly specific issue seen in production, we could use a little bit of comments about the mechanics.